lib/dockerfile-parser.ts

```diff
@@ -0,0 +1,115 @@
+import { CommandEntry, parse } from 'docker-file-parser';
+
+const LiveCommandDirective = 'dev-cmd-live';
+const EscapeDirective = 'escape';
+
+export function parseDockerfile(content: string | Buffer): CommandEntry[] {
+	// This function may look a little weird, but due to bugs
+	// in the comment parsing of docker-file-parser we first go
+	// through the dockerfile, find out directives, and keep a
+	// track of the line number that it was encountered at.
+	// We then pass to the docker-file-parser parsing
+	// function, with no comments. We then integrate the
+	// directive into the returned values using the line
+	// number (as the order matters).
+
+	// issue with comments in docker-file-parser:
+	// https://github.com/joyent/node-docker-file-parser/issues/8
+
+	if (Buffer.isBuffer(content)) {
+		content = content.toString();
+	}
+
+	// We keep track of this, as one of the directives can
+	// change it
+	let escapeCharacter = '\\';
+	let lastNonCommentLine = '';
+
+	const directives: CommandEntry[] = [];
+	const nonCommentLines = content
+		.split(/\r?\n/)
+		.map((line, idx) => {
+			line = line.trim();
```

Comment:
The space might have a meaning:
```
RUN echo 'def some_python_function():\n\
    print("Hola!")' > some_file.py
```

Reply:
@zvin completely agree, but this only checks for lines beginning with `#`, anything else is left to `docker-file-parser` to handle (which does handle this fine). I've added a test for this anyway to make sure it doesn't break.

Reply:
I must be missing something @CameronDiver .
You trim the line with `line = line.trim()` then you return it line 59 then you join all lines line 61 and give this string to docker-file-parser.

Reply:
That is true... I'll work out what's going on first :P

Reply:
@zvin the escaping in the test was wrong, I was escaping the newline, so the spaces were then not leading. I've fixed the test and the code now.